### ISSUE TITLE
[Ubuntu backports] Updating to Cockpit 329 removes cockpit-pcp

### ISSUE THREAD

**Issue by martinpitt** [cockpit-project/cockpit#21344]
### Discussed in https://github.com/cockpit-project/cockpit/discussions/21343

<div type='discussions-op-text'>

<sup>Originally posted by **compbl** November 27, 2024</sup>
When I attempt to update to Cockpit 329 via the software updates page I am receiving an error "The following packages have unmet dependencies:  cockpit-bridge: Conflicts: cockpit-pcp"  I am assuming I have to remove cockpit-pcp, but cannot find how to do that.   Any help would be appceiated.   Running on Ubuntu 22.04.5 LTS

Thanks</div>

**Comment by emaayan** [comment]
so do i need to remove cokcpit-pcp fresh installation ?  just do apt install cockpit cockpit-sos  and that's it? cause right now i don't get pcp

**Comment by martinpitt** [comment]
@emaayan Not sure what you mean, but with recent versions there is no cockpit-pcp, right. If you want historic metrics, install `python3-pcp`. The metrics page will also do this for you with on-demand installation.

**Comment by emaayan** [comment]
> @emaayan Not sure what you mean, but with recent versions there is no cockpit-pcp, right. If you want historic metrics, install `python3-pcp`. The metrics page will also do this for you with on-demand installation.

well i don't install cockpit-pcp i get a banner in the performance tab that asks me if i want to install it.

**Comment by martinpitt** [comment]
@emaayan Can you please give me the output of `dpkg -l '*cockpit*'`?

**Comment by emaayan** [comment]
[image: image.png]

**Comment by emaayan** [comment]
ok i think i see the problem..

**Comment by martinpitt** [comment]
Heh, ok -- attaching images via email is apparently not working :sweat_smile:

**Comment by emaayan** [comment]
good it would be less embarrassing cause at the time i didn't catch you previous reply about the cockpit itself not being pinned so now i have cockpit 14 and the rest of the cockpits at 3.29.1

**Comment by emaayan** [comment]
still a problem, and missing pcp
![image](https://github.com/user-attachments/assets/f2a28482-284e-40ff-bf02-9f0185efabe2)

![image](https://github.com/user-attachments/assets/4d001d3f-0f82-4fbe-bad8-4712d70828bc)

**Comment by LeLawnGames** [comment]
Any fixes for this yet? My existing cockpit instance is unable to have updates installed because of the cockpit-pcp error.

**Comment by martinpitt** [comment]
@LeLawnGames are `python3-pcp` and `pcp` installed? These are what "Install PCP support" should do.

**Comment by emaayan** [comment]
> @LeLawnGames are `python3-pcp` and `pcp` installed? These are what "Install PCP support" should do.

isn't there a way to install them from the start?

**Comment by amatson97** [comment]
I had a similar issue on my install, I was able to clear the failing services by running:

sudo apt-get upgrade pcp
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Calculating upgrade... Done
The following NEW packages will be installed:
  libpcp-pmda-perl libpcp-trace2 libpcp-web1 libpfm4 pcp

Not sure if this helped, but i did enable both of this and install which also seemed to install some additional packages.

![Screenshot from 2024-12-01 18-01-56](https://github.com/user-attachments/assets/9b9afa4d-d4b3-474e-a09a-c5cc7e82257f)

**Comment by LeLawnGames** [comment]
Yeah it's the above error I keep encountering -- Conflicts: cockpit-pcp

**Comment by LunarTwilight** [comment]
I was a somewhat similar issue with no historical logs, installing `pcp` helped, as it was uninstalled by upgrading.

**Comment by berwhale** [comment]
`sudo apt remove cockpit-pcp` fixed the issue for me.

**Comment by ShiftedRy** [comment]
> `sudo apt remove cockpit-pcp` fixed the issue for me. I had to reboot afterwards to get the metrics back.

worked for me too!

**Comment by LeLawnGames** [comment]
Yep worked for me!

**Comment by kevinceegross** [comment]
`sudo apt remove cockpit-pcp` fixed the issue for me as well.  Thank you!

**Comment by emaayan** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/13434544?v=4&amp;size=80">@emaayan](https://github.com/emaayan) Not sure what you mean, but with recent versions there is no cockpit-pcp, right. If you want historic metrics, install `python3-pcp`. The metrics page will also do this for you with on-demand installation.

is there any update on resolving this with current existing versions who do still have cockpit installed? without using a cli?

**Comment by oparada1988** [comment]
> > [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/13434544?v=4&amp;size=80">@emaayan](https://github.com/emaayan) Not sure what you mean, but with recent versions there is no cockpit-pcp, right. If you want historic metrics, install `python3-pcp`. The metrics page will also do this for you with on-demand installation.
> 
> is there any update on resolving this with current existing versions who do still have cockpit installed? without using a cli?

Bump, I would like to use the newest release, but pcp issue persist. I have python3-pcp and pcp installed with no avail

**Comment by martinpitt** [comment]
So apt can do the upgrade across the obsolete cockpit-pcp replacement, PackageKit can't. I'm afraid that's outside of what Cockpit can influence 😢

**Comment by geniot** [comment]
I also had some unmet dependencies when trying to update through the web ui. 
Then I ran on the server:
`sudo apt-get upgrade pcp`
`sudo apt-get upgrade cockpit`
And cockpit reported that the system is up to date.
